btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+        # Delays and edges should be the same length if error checking in the
+        # process_calib_run function works properly
+        #@todo implement amplitude- and fwhm-based selection and compare to this
+        # simplified version
+        self.edges = edges
+        self.delays = delays
+        if (edges > 250).any():
+            delays_fit = delays[edges > 250]
+            edges_fit = edges[edges > 250]
+            if (edges_fit < 870).any():
+                delays_fit = delays_fit[edges_fit < 870]
+                edges_fit = edges_fit[edges_fit < 870]
+                inrange = True
+
+                # For testing
+                self.edges_fit = edges_fit
+                self.delays_fit = delays_fit
+        if inrange:
+            print('Fitting calibration between 250 and 870 pixels.')
+            self._model = np.polyfit(edges_fit, delays_fit, order)
+        else:
+            print('Fit data is out of 250-870 pixel range. Results questionable.')
+            self.edges_fit = self.edges
+            self.delays_fit = self.delays
+            self._model = np.polyfit(edges, delays, order)
+
+    def ttstage_code(self, hutch: str) -> str:
+        """! Return the correct code for the time tool delay (in ns) for a given
+        experimental hutch.
+
+        @param hutch (str) Three letter hutch name. E.g. mfx, cxi, xpp
+
+        @return code (str) Epics code for accessing hutches time tool delay.
+        """
+        h = hutch.lower()
+        num = ''
+
+        if h == 'xpp':
+            num = '3'
+        elif h == 'xcs':
+            num = '4'
+        elif h == 'mfx':
+            num = '45'
+        elif h == 'cxi':
+            num = '5'
+
+        if num:
+            return f'LAS:FS{num}:VIT:FS_TGT_TIME_DIAL'
+        else:
+            raise InvalidHutchError(hutch)
+
+    def edge_to_time(self, edge_pos: int) -> float:
+        """! Given an internal model and an edge position, return the time tool
+        jitter correction.
+
+        This method is called by the partner method actual_time; however, it is
+        provided directly so that users who have a different time convention
+        (i.e. negative times when pump arrives before xray) can more easily
+        correct their data.
+
+        @param edge_pos (int | array-like) Position (pixel) of detected edge on camera.
+
+        @return correction (float) Jitter correction in picoseconds.
+        """
+        correction = 0
+        if len(self._model) > 0:
+            n = len(self._model)
+            for i, coeff in enumerate(self._model):
+                correction += coeff*edge_pos**(n-i-1)
+            print('Using model to correct for jitter.')
+        else:
+            print('No calibration model. Jitter correction not possible.')
+
+        ## LAS:FS:VIT:FS... is in NANOSECONDS. Multiply model fit by 1000 to get
+        # the correction in ps. MAKE SURE YOUR DIRECTION OF TIME IS CORRECT. YOU
+        # MAY NEED A FACTOR OF -1!
```

Comment:
How do we check time direction is correct? Sounds like an issue to me :D

Reply:
I chose the standard convention. There may be some groups which reverse the order of positive time (e.g. positive time is x-rays arriving first). This was a note to myself more than anything, but it's worth keeping in mind if it comes up in an experiment.

We can potential put an optional parameter to multiply by -1 to address this.